src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+public class FigureSupplier {
+    private final List<FigureType> availableFigures = Arrays.asList(FigureType.values());
+    private final Random random = new Random();
+    private final ColorSupplier color = new ColorSupplier();
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, Color.WHITE);
+    }
+
+    public Figure getRandomFigure(boolean randomParameters) {
+        int figuresSize = availableFigures.size();
+        int randomIndex = random.nextInt(figuresSize - 1);
```

Comment:
The `random.nextInt(figuresSize - 1)` call should be `random.nextInt(figuresSize)` to ensure that the last figure type in the list can also be selected. The current implementation excludes the last figure type.